sparse_dot_topn/sparse_dot_topn_source.cpp

```diff
@@ -27,105 +29,418 @@
 bool candidate_cmp(candidate c_i, candidate c_j) { return (c_i.value > c_j.value); }
 
 /*
-    C++ implementation of sparse_dot_topn
+	C++ implementation of sparse_dot_topn
 
-    This function will return a matrxi C in CSR format, where
-    C = [sorted top n results and results > lower_bound for each row of A * B]
+	This function will return a matrix C in CSR format, where
+	C = [sorted top n results > lower_bound for each row of A * B]
 
-    Input:
-        n_row: number of rows of A matrix
-        n_col: number of columns of B matrix
+	Input:
+		n_row: number of rows of A matrix
+		n_col: number of columns of B matrix
 
-        Ap, Aj, Ax: CSR expression of A matrix
-        Bp, Bj, Bx: CSR expression of B matrix
+		Ap, Aj, Ax: CSR expression of A matrix
+		Bp, Bj, Bx: CSR expression of B matrix
 
-        ntop: n top results
-        lower_bound: a threshold that the element of A*B must greater than
+		ntop: n top results
+		lower_bound: a threshold that the element of A*B must greater than
 
-    Output by reference:
-        Cp, Cj, Cx: CSR expression of C matrix
+	Output by reference:
+		Cp, Cj, Cx: CSR expression of C matrix
 
-    N.B. A and B must be CSR format!!!
+	N.B. A and B must be CSR format!!!
 */
-void sparse_dot_topn_source(int n_row,
-                        int n_col,
-                        int Ap[],
-                        int Aj[],
-                        double Ax[], //data of A
-                        int Bp[],
-                        int Bj[],
-                        double Bx[], //data of B
-                        int ntop,
-                        double lower_bound,
-                        int Cp[],
-                        int Cj[],
-                        double Cx[])
+void sparse_dot_topn_source(
+		int n_row,
```

Comment:
Why converting spaces indentation to tabs?

Reply:
That's my personal preference.  I didn't think anyone would mind.  Is it causing problems with your code-style analyzer?

Reply:
Most library I know use spaces, I just wanted to check if it was on purpose since the diffs are less nice, but ok